tracer/src/Datadog.Trace/ClrProfiler/CallTarget/Handlers/Continuations/ContinuationGenerator.cs

```diff
@@ -34,5 +39,27 @@ protected static TTo FromTReturn<TTo>(TReturn returnValue)
             return ContinuationsHelper.Convert<TReturn, TTo>(returnValue);
 #endif
         }
+
+        internal abstract class ContinuationResolver
```

Comment:
naming nit: These couple of Generator and Resolver classes are (understandably) complex and kind of hard to reason about for people unfamiliar with the code, so I was thinking that with some renaming we might be able to make it a little easier to understand. We don't have to do these renames in this PR, but I had some ideas for possible renames:
- Changing `ContinuationResolver` to `CallbackHandler`, since our `OnAsyncMethodEnd` method is kind of a "callback" and the various subclasses handle the execution for the different return types
- Changing method name `SetContinuation` to `ExecuteContinuation`/`ExecuteCallback`, since we will always execute our `OnAsyncMethodEnd` method (either synchronously or asynchronously) and return a result

What do you think about these suggestions?

Reply:
I wen with `CallbackHandler` in [79f1332](https://github.com/DataDog/dd-trace-dotnet/pull/3555/commits/79f1332071c1b502de0d0a058d9bd942081a5841)